src/Microsoft.HttpRepl/FileSystem/RealFileSystem.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.HttpRepl.FileSystem
+{
+    internal class RealFileSystem : IFileSystem
+    {
+        public bool FileExists(string path)
+        {
+            return File.Exists(path);
+        }
+
+        public void WriteAllTextToFile(string path, string contents)
+        {
+            VerifyDirectoryExists(path);
+            File.WriteAllText(path, contents);
+        }
+
+        public byte[] ReadAllBytesFromFile(string path)
+        {
+            return File.ReadAllBytes(path);
+        }
+
+        public string[] ReadAllLinesFromFile(string path)
+        {
+            return File.ReadAllLines(path);
+        }
+
+        public void WriteAllLinesToFile(string path, IEnumerable<string> contents)
+        {
+            VerifyDirectoryExists(path);
+            File.WriteAllLines(path, contents);
+        }
+
+        public void DeleteFile(string path)
+        {
+            File.Delete(path);
+        }
+
+        public string GetTempFileName()
+        {
+            return Path.GetTempFileName();
+        }
+
+        private void VerifyDirectoryExists(string path)
+        {
+            string directoryName = Path.GetDirectoryName(path);
+            if (!Directory.Exists(directoryName))
```

Comment:
nit: It's safe to call Directory.Create if the directory already exists, it'll noop.

Reply:
Gotcha. Simplified in c6d4a44